pkg/doc/verifiable/credential.go

```diff
@@ -316,12 +381,133 @@ func NewCredential(dataJSON []byte, opts ...CredentialOpt) (*Credential, error)
 	return cred, nil
 }
 
+// CredentialClaims is JWT Claims extension by Credential.
+type CredentialClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// RawClaims is used to get raw content of "vc" claim of JWT.
+type RawClaims struct {
+	*jwt.Claims
+
+	Raw map[string]interface{} `json:"vc,omitempty"`
+}
+
+func decodeJws(rawJwt []byte, toVerifyJwtSignature bool, fetcher PublicKeyFetcher) ([]byte, *rawCredential, error) {
+	parsedJwt, err := jwt.ParseSigned(string(rawJwt))
+	if err != nil {
+		return nil, nil, fmt.Errorf("VC is not JWS: %w", err)
+	}
+
+	var jwtPayload CredentialClaims
+	err = parsedJwt.UnsafeClaimsWithoutVerification(&jwtPayload)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse JWT claims: %w", err)
+	}
+
+	if toVerifyJwtSignature {
```

Comment:
Why would we **not** want to verify the signature?

Reply:
The reason is to not to do it twice when using VC extension. We first encode into the base data model (with signature verification) and check what's the extension is it (e.g. by `@context` and `type`) and then make another encoding but into the concrete data model extension. At the latter stage, we do not need to verify the signature once again.

Reply:
@llorllale @troyronda After talking to Georgy, we've decided for simplification to always verify the signature.